src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs

```diff
@@ -16,29 +17,46 @@ public class GivenThatWeWantToBuildWithGlobalJson : SdkTest
         public GivenThatWeWantToBuildWithGlobalJson(ITestOutputHelper log) : base(log)
         {}
 
-        [Fact]
-        public void It_fails_build_on_failed_sdk_resolution()
+        [FullMSBuildOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_fails_build_on_failed_sdk_resolution(bool runningInVS)
         {
-            var fakePath = "fakePath";
             TestProject testProject = new TestProject()
             {
                 Name = "FailedResolution",
                 IsSdkProject = true,
                 TargetFrameworks = "net5.0"
             };
-            testProject.AdditionalProperties["SdkResolverHonoredGlobalJson"] = "false";
-            testProject.AdditionalProperties["SdkResolverGlobalJsonPath"] = fakePath;
 
             var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var globalJsonPath = Path.Combine(testAsset.Path, testProject.Name, "global.json");
+            File.WriteAllText(globalJsonPath, @"{
+""sdk"": {
+""version"": ""9.9.999""
+}
+}");
 
-            var buildCommand = new BuildCommand(testAsset);
-            buildCommand.Execute()
+            var resolveSdkCommand = new ResolveSdkCommand(testAsset);
+            var result = resolveSdkCommand.Execute($"/p:BuildingInsideVisualStudio={runningInVS}")
                 .Should()
-                .Fail()
-                .And
-                .HaveStdOutContaining("NETSDK1141")
-                .And
-                .HaveStdOutContaining(fakePath);
+                .Fail();
+            var warningString = "warning : Unable to locate the .NET SDK as specified by global.json, please check that the specified version is installed.";
```

Comment:
I think it's also supposed to fail to build in VS.  It should load successfully, and design-time builds should succeed, but you should still get a build error.  Here's the spec:  https://github.com/dotnet/designs/blob/master/accepted/2020/workloads/workload-resolvers.md#globaljson-failures

Reply:
Yes, it does fail to build (see line 43). I'm asserting that we see the correct error/warning messages from the resolver here but I've been unable to assert the build error because we fail before that erroring target is run when we're outside of VS because we're missing the ResolveSdk target.